### DISCUSSION TITLE
I can not create LTSP image

### DISCUSSION THREAD

**Discussion by isobelle2998** [ltsp/ltsp#254] [Q&A]
I tried to create an image using a image with the .qcow2 extension.
The image runs well.
Primarly when I tried to run this command: 
`ln -rs home/ltsp-server/Téléchargements/lubuntu.qcow2 /srv/ltsp/lubuntu.img`, 
I was told that the directory `/srv/ltsp` does not exist. So I had to create ltsp in /srv myself
Then, when I run this command:
`ltsp image debian`
I get this error:
```
LTSP command failed: blkid -po export /srv/ltsp/lubuntu.img
Aborting ltsp
I don't know how to mount /srv/ltsp/lubuntu.img
```

**Comment by alkisg** [comment]
LTSP doesn't support .vdi or .qcow2 or other compressed / custom images.
It only supports -flat.vmdk or .raw or .dd images, that contain the raw sectors of the virtual disk.
That file can be sparse, to save space (google for sparse files).
https://ltsp.org/man/ltsp-image/#vm-images describes how to create these from virtualbox; a similar method should exist for virt-manager etc.

So, for example, check the correct output of the following commands:

```shell
# file -L /srv/ltsp/x86_32.img
/srv/ltsp/x86_32.img: DOS/MBR boot sector
# blkid -p /srv/ltsp/x86_32.img
/srv/ltsp/x86_32.img: PTUUID="4bc5f12c" PTTYPE="dos"
```

**Comment by alkisg** [comment]
I transferred this issue to discussions (https://github.com/ltsp/community/issues) instead of bugs (https://github.com/ltsp/ltsp/issues).

**Comment by isobelle2998** [comment]
This is the output of the commands above:
```
root@ltspserver-Standard-PC-Q35-ICH9-2009:~# file -L /srv/ltsp/x86_32.img
/srv/ltsp/x86_32.img: cannot open `/srv/ltsp/x86_32.img' (No such file or directory)
root@ltspserver-Standard-PC-Q35-ICH9-2009:~# blkid -p /srv/ltsp/x86_32.img
blkid: erreur : /srv/ltsp/x86_32.img: Aucun fichier ou dossier de ce type
```

**Comment by alkisg** [comment]
You need to use YOUR file (/srv/ltsp/lubuntu.img), not my example (/srv/ltsp/x86_32.img).
E.g.

```shell
file -L /srv/ltsp/lubuntu.img
```

Which software are you using, virt-manager?

**Comment by isobelle2998** [comment]
Yes I am using virt-manager.
These are the outputs:
```
root@ltspserver-Standard-PC-Q35-ICH9-2009:~# file -L /srv/ltsp/lubuntu.img
/srv/ltsp/lubuntu.img: QEMU QCOW Image (v3), 8589934592 bytes
root@ltspserver-Standard-PC-Q35-ICH9-2009:~# blkid -p /srv/ltsp/lubuntu.img
root@ltspserver-Standard-PC-Q35-ICH9-2009:~#
```

**Comment by alkisg** [comment]
Right, so " QEMU QCOW Image" is the wrong format.
Delete that image, and create a new one, with format "raw", not "qcow2".

**Comment by isobelle2998** [comment]
I tried with the format "raw". And it worked fine. Thank you.

**Comment by luxzg** [comment]
> LTSP doesn't support .vdi or .qcow2 or other compressed / custom images. It only supports -flat.vmdk or .raw or .dd images, that contain the raw sectors of the virtual disk.

Thank you for the info, I just wanted to check that it's still complete & current info.

https://ltsp.org/man/ltsp-image/ says:

`you may use VirtualBox, virt-manager, KVM, VMWare and similar tools to maintain one or more template images for the clients`

From that page it sounds as if way more than vmdk/raw/dd is supported.

May I suggest adding list of exact supported image formats to that page?

**Comment by alkisg** [comment]
That page also says:

> At the disk creation dialog, select "VMDK" type and "Fixed size", not "Dynamically allocated".

and:

> In summary, you may symlink **raw** VM disks in /srv/ltsp/img_name.img

That's really the whole list there, just "raw", no other formats are supported. They may be named .raw or -flat.vmdk or .dd or even be symlinks to disks or partitions (/dev/sdXY), but all of them are the same thing, plain raw file systems that `losetup` and `mount` can access. This is because these two are the only tools available in the initramfs, when the clients boot; there are no tools to mount .vdi or .qcow2 images available in the initramfs.
